project-fob/Controllers/HostController.cs

```diff
@@ -23,6 +24,41 @@ public ActionResult Index()
         {
             return View();
         }
+
+        public ActionResult QrCode()
+        {
+            var gotvalue = HttpContext.Session.TryGetValue("meetingid", out var meetingIdValue);
+            string MeetingIdString = System.Text.Encoding.ASCII.GetString(meetingIdValue);
+
+
+            string baseUrl = /*Request.Url.Scheme*/ Request.GetDisplayUrl(); // + "://" + Request.Url.Authority + Request.ApplicationPath.TrimEnd('/') + "/";
+            int count = baseUrl.Count(f => f == '/');
+            string[] split = baseUrl.Split('/');
+            StringBuilder sb = new StringBuilder();
+
+            for (int i = 0; i < split.Length-2; i++)
+            {
+                sb.Append(split[i]);
+            }
+            Meeting meet = db.Meeting.Single(x => x.MeetingId.Equals(MeetingIdString));
+
+            @ViewBag.url = sb.ToString() + "/Home/meetingPageUser?meetingId=" + MeetingIdString + "&password=" + meet.RoomPassword;
+            return View("~/Views/Home/QRCode.cshtml");
+        }
+
+        protected void Button1_Click(object sender, EventArgs e)
```

Comment:
`Button1` is an icky name for a button, it would be better if it was named something more identifiable.

Reply:
This was an accident. whoops.